store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreCache.java

```diff
@@ -25,16 +25,22 @@
 import java.util.Set;
 
 import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
 
 /**
  * Wrapper around the {@link uk.gov.gchq.gaffer.cache.CacheServiceLoader} to provide an interface for
  * handling the {@link Graph}s within a {@link uk.gov.gchq.gaffer.federatedstore.FederatedStore}.
  */
 public class FederatedStoreCache extends Cache<Pair<GraphSerialisable, FederatedAccess>> {
     public static final String ERROR_ADDING_GRAPH_TO_CACHE_GRAPH_ID_S = "Error adding graph to cache. graphId: %s";
+    private static final String CACHE_SERVICE_NAME_PREFIX = "federatedStoreGraphs";
 
     public FederatedStoreCache() {
-        super("federatedStoreGraphs");
+        this(null);
+    }
+
+    public FederatedStoreCache(final String cacheNameSuffix) {
+        super(String.format("%s%s", CACHE_SERVICE_NAME_PREFIX, nonNull(cacheNameSuffix) ? "_" + cacheNameSuffix.toLowerCase() : ""));
```

Comment:
I like what this is doing despite it being somewhat hard to read combining .format and ternary

Reply:
Would be better split over multiple lines. There are library methods which could help with the non-null handling.